doc/tutorial/single_cell_detailed_recipe.rst

```diff
@@ -3,127 +3,82 @@
 The recipe
 **********
 
-The :class:`arbor.single_cell_model` of the previous example created a :class:`arbor.recipe` under
+The :class:`arbor.single_cell_model` of the original example created a :class:`arbor.recipe` under
 the hood, and abstracted away the details so we were unaware of its existence. In this example, we will
 examine the recipe in detail: how to create one, and why it is needed.
 
 .. literalinclude:: ../../python/example/single_cell_detailed_recipe.py
    :language: python
-   :lines: 91-139
+   :lines: 84-120
 
 Let's go through the recipe point by point.
 
-Step **(6)** creates a ``single_recipe`` class that inherits from :class:`arbor.recipe`. The base recipe
-implements all the methods defined above with default values except :meth:`arbor.recipe.num_cells`,
-:meth:`arbor.recipe.cell_kind` and :meth:`arbor.recipe.cell_description` which always have to be implemented
-by the user. The :meth:`arbor.recipe.global_properties` also needs to be implemented for
-:class:`arbor.cell_kind.cable` cells. The inherited recipe can implement any number of additional methods and
-have any number of instance or class variables.
+Step **(5)** creates a ``single_recipe`` class that inherits from :class:`arbor.recipe`.
+:meth:`arbor.recipe.num_cells`, :meth:`arbor.recipe.cell_kind` and :meth:`arbor.recipe.cell_description`
+always have to be implemented by the user. We'll also implement :meth:`arbor.recipe.global_properties` to be able
+to decorate :class:`arbor.cell_kind.cable` cells with mechanisms and :meth:`arbor.recipe.probes` to be able to
+insert the probe.
 
-Step **(6.1)** defines the class constructor. In this case, we pass a ``cell`` and a set of ``probes`` as
-arguments. These will be used to initialize the instance variables ``self.the_cell`` and ``self.the_probes``,
-which will be used in the overloaded ``cell_description`` and ``get_probes`` methods. Before variable
-initialization, we call the base C++ class constructor ``arbor.recipe.__init__(self)``. This ensures correct
-initialization of memory in the C++ class.
+Step **(5.1)** defines the class constructor. As per :class:`arbor.recipe` instructions, we call
+``arbor.recipe.__init__(self)`` to ensure correct initialization of memory in the C++ class.
 
-We also create the ``self.the_cat`` variable and set it to arbor's default mechanism catalogue. This will expose
-the *hh* and *pas* mechanisms but not the *Ih* mechanism, which is present in the allen catalogue. To be able
-to use *Ih*, we extend ``self.the_cat`` to include the allen catalogue.
+We also create an instance variable for the mechanism catalogue, the ``self.the_cat`` variable. This ensures
+its lifetime is guaranteed to extend to the entire duration of the simulation. Setting the mechanism catalogue
+in the constructor is a pattern you'll see return in these tutorials.
+
+We set ``self.the_cat`` to Arbor's default mechanism catalogue. This will expose the *hh* and *pas* mechanisms but not the *Ih* mechanism,
```

Comment:
This is no longer needed in the code. You can just write `self.the_props.catalogue.extend(...)`.
Also it has been already modernised, this is just outdated ;)

Reply:
In future, someone writing new features might consider a pass by the examples then ;)